### PR TITLE
Refactor code for effect of N

### PR DESCRIPTION
**Pull request by dmcglinn**
A lot of work still needs to be done. These changes primarily target the analysis related to the effect of N.

### CODE REVIEW

File: R/mobr.R [hunk 1]
```diff
@@ -214,32 +212,39 @@ rarefaction = function(x, method, effort=NULL) {
...
-  effort = effort[which(effort <= min(sum(comm_group), ref_dens * nrow(comm_group)))]
-  interp_S_samp_rare = pchip(c(1, ref_dens * (1:nrow(comm_group))), c(1, S_samp_rare), effort)
-  S_indiv_rare = rarefaction(group_sad, 'indiv', effort = effort)
-  deltaS = interp_S_samp_rare - as.numeric(S_indiv_rare)
-  out = data.frame(cbind(effort, deltaS))
-  names(out) = c('effort', 'deltaS')
+# Auxillary function: difference between the ind-based rarefaction and the
+# sample-based rarefaction for one group with the evaluation sample size (number
+# of individuals) defined by ref_dens, evaluated at specified points (given by
+# "effort") New input: permute - if permute is FALSE, S_samp_rare is generated
```
Comment (rueuntal) [inline R/mobr.R]: I don't see "permute". Is it dropped?

File: R/mobr.R [hunk 2]
```diff
@@ -214,32 +212,39 @@ rarefaction = function(x, method, effort=NULL) {
...
+  # as each of the nplots are collected
+  samp_effort = round((1:nplots) * group_dens)
+  #effort = effort[which(effort <= min(sum(comm_group), ref_dens * nrow(comm_group)))]
+  # use individual based rarefaction evaluated at the 
+  # number of individuals sampled for each plot
+  S_samp = rarefaction(comm_group, 'indiv', samp_effort)
+  # rescale and interpolate this plot based S to individual based 
+  # using the ref_density (i.e., not the observed density)
+  rescaled_effort = round(1:nplots * ref_dens)
+  interp_S_samp = pchip(c(1, rescaled_effort),
```
Comment (rueuntal) [inline R/mobr.R]: Would there be any problem if `max(inds)` is larger than `max(rescaled_effort)` (ie extrapolation)?

File: R/mobr.R [hunk 3]
```diff
@@ -268,32 +273,85 @@ rarefy_sample_explicit = function(comm_one_group, xy_one_group) {
...
-    comm_group = comm[row_indices, ]
-    sp_sums = colSums(comm_group)
-    tmp_comm = sapply(sp_sums, function(x) 
-      table(c(sample(1:nrow(comm_group), x,
-                     replace=T),
-              1:nrow(comm_group))) - 1)
-    comm_group_noagg[row_indices, ] = tmp_comm
-  }  
-  comm_group_noagg
+#' Permute community matrix within groups
```
Comment (rueuntal) [inline R/mobr.R]: The rewriting of these functions is great!

**Comment by rueuntal** [comment]
Looking great! Thanks for the major rewrite. If the extrapolation using `pchip` is not an issue I'll go ahead and merge it.

File: R/mobr.R (hunk 1 above)
Comment (dmcglinn) [inline R/mobr.R]: Yea that argument is no longer needed. I think I added that documentation before I realized that we only needed individual based rarefaction this needs to be removed.

File: R/mobr.R (hunk 2 above)
Comment (dmcglinn) [inline R/mobr.R]: Yes you are right that would be a problem.

**Comment by dmcglinn** [comment]
I'm happy to address the problems you noted in the morning, but if you'd rather get working with this version of the code and make these minor changes yourself feel free. Either way is good for me.

**Comment by dmcglinn** [comment]
I should note that I didn't test out the `get_delta_stats` function in its entirety. It is possible (likely) that I broke a few small parts while trying to rip the guts out it and put it back together again.

**Comment by rueuntal** [comment]
Sounds good. We can leave testing and fixing small issues for later.

**Pull request merged**